adam/perception/perception_graph_nodes.py

```diff
@@ -98,6 +98,41 @@ def dot_label(self):
         )
 
 
+@attrs(slots=True, frozen=True, eq=False)
+class Size(GraphNode):
+    x: float = attrib(validator=instance_of(float))
+    y: float = attrib(validator=instance_of(float))
+    area: float = attrib(validator=instance_of(float))
+
+    def dot_label(self) -> str:
+        return f"Size(x={self.x}, y={self.y}, area={self.area})"
+
+
+@attrs(slots=True, frozen=True, eq=False)
+class RelativeSize(GraphNode):
+    x_bigger_than: float = attrib(validator=instance_of(bool))
+    y_bigger_than: float = attrib(validator=instance_of(bool))
+
+    def dot_label(self) -> str:
+        return (
+            f"RelativeSize(x_bigger_than={self.x_bigger_than},"
+            f" y_bigger_than={self.y_bigger_than})"
+        )
```

Comment:
Rather than a new node for `RelativeSize` we should consider re-using the size representation from Phase 1/2. Essentially that was just an edge for `BiggerThan(Object_X, Object_Y)` where it was asserted that Object_X was the larger object. If we reuse the other representation, we won't need to build new predicates (and given my understanding is the information contained is the same it should be fine? @spigo900 Do you agree?)

Reply:
@lichtefeld Yes, I agree, I think this makes sense.

Reply:
The relative size features that we decided to extract in the pre-processing phase refer specifically to the relative widths and heights of the bounding boxes around each object. From what I'm reading, it doesn't look like the `biggerThan` and `smallerThan` edges support this (see [this issue](https://github.com/isi-vista/adam/issues/70))

Reply:
You're right, I forgot about that. In that case this representation makes sense to me.

Reply:
I had elected to represent the other relative feature-set (`relative_distance`) as a mapping of object IDs to nodes; should I switch to representing it with edges (instead of nodes) as well? I also realized we never use the absolute `distance` feature in the perception graph--it just gets generated by pre-processing and then never used by ADAM.

Reply:
The mapping setup is fine for this intermediate representation. In the graph relative distance should be a node plus a pair of edges, e.g. for a toy car on a table, something like `table --(in-region)--> relative_distance(x_offset: 0, y_offset=5, ...) --(reference-object)-->toy_car`. This is similar to how `Region` works. This is what i meant about adding the relative distance information to the graph.

Reply:
I'm fine with `relative_distance` being a node, it should have connections to the two referenced objects as edges in the graph then. I'm forgetting the exact edge name but its something like `reference object`? Our current `relative_distance` is just the shortest line between the two objects correct? We don't break it out in to the component X/Y axes do we? No is fine, but if we are separating on the per-axis basis then each of the `relative_distance` relationships should get separate nodes with a label indicating if its an `X` or `Y` axis relative distance.

With respect to the absolute `distance` value. It should probably be encoded so it can be presented as output but I don't see how an absolute value is useful for learning concepts, even if its important in understanding and analyzing the specific instance.

Reply:
@lichtefeld `reference object` sounds right. I believe we are calculating the axis distances as well as straight-line. I agree it is probably simplest to break it out by axis/label -- x or y -- especially since then you can just use a continuous node to represent it.

I agree we probably don't need the straight-line distance, no.